File: test-framework/src/main/java/api/app/astrodao/com/core/dto/api/proposals/Role.java
package api.app.astrodao.com.core.dto.api.proposals;

import lombok.Data;

import java.util.List;

@Data
public class Role<T> {
    private String createdAt;
    private String id;
    private String name;
    private T kind;
    private Object balance;
    private Object accountIds;
    private List<String> permissions;
    private VotePolicy votePolicy;
}

File: libs/dao/src/types/role.ts
import { VotePolicy } from './vote-policy';
import { RoleKindType } from '../entities';

export class RolePermission {
  name: string;
  kind: RoleKindType;
  balance?: number;
  accountIds?: string[];
  permissions: string[];
  votePolicy: { [key: string]: VotePolicy };
}
